compiler/qsc_rca/src/core.rs

```diff
@@ -684,12 +696,24 @@ impl<'a> Analyzer<'a> {
                 application_instance.get_expr_compute_kind(e).is_dynamic()
             });
         if is_any_sub_expr_dynamic {
-            let dynamic_value_kind = ValueKind::new_dynamic_from_type(expr_type);
-            let dynamic_runtime_features =
+            let dynamic_value_kind = if matches!(expr_type, Ty::Array(..)) {
+                // An array coming from a dynamic conditional should be treated as dynamic in length
+                // and content.
+                ValueKind::Array(RuntimeKind::Dynamic, RuntimeKind::Dynamic)
+            } else {
+                ValueKind::new_dynamic_from_type(expr_type)
+            };
+            let mut dynamic_runtime_features =
                 derive_runtime_features_for_value_kind_associated_to_type(
                     dynamic_value_kind,
                     expr_type,
                 );
+            if is_any_result(expr_type) {
+                dynamic_runtime_features |= RuntimeFeatureFlags::UseOfDynamicResult;
+            }
+            if matches!(expr_type, Ty::Tuple(tup) if !tup.is_empty()) {
+                dynamic_runtime_features |= RuntimeFeatureFlags::UseOfDynamicTuple;
+            }
```

Comment:
There's a bug here... this is rejecting use of the `Measure` operation, even though the condition there is classical.

Reply:
Whew, found it!